### PR TITLE
Ensure bash version at least 5.x

### LINKED ISSUE #98294: build kubernetes binaries failed
**What happened**:
```
joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS=""
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS="-s -w"
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 
build/../build/common.sh: line 579: conditional binary operator expected
```
**What you expected to happen**:

build kubernetes binaries successfully

**How to reproduce it (as minimally and precisely as possible)**:
```
joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS=""
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 GOLDFLAGS="-s -w"
build/../build/common.sh: line 579: conditional binary operator expected

joshua:kubernetes joshua$ build/run.sh make all KUBE_BUILD_PLATFORMS=linux/amd64 
build/../build/common.sh: line 579: conditional binary operator expected
```

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`):  k8s v1.18.15+ , k8s v1.19.7 , k8s v1.20.2+
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):   my host environment: MacOS
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by dims**
Reduce friction by explicitly telling folks that they need newer versions of bash.

Example we use `-v`:
```test -v N' can now test whether or not positional parameter N is set.```

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #98294

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/lib/util.sh [hunk 1]
```diff
@@ -729,6 +729,16 @@ function kube::util::ensure_dockerized {
   fi
 }
 
+# kube::util::ensure-bash-version
+# Check if we are using a supported bash version
+#
+function kube::util::ensure-bash-version {
+  if ((${BASH_VERSINFO[0]}<5)); then
+    echo "ERROR: This script needs at bash version at least 5.x"
```
Comment (justaugustus) [inline hack/lib/util.sh:737]:
```suggestion
    echo "ERROR: This script requires a minimum bash version of 5.0"
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
@dims -- Small language nit

File: hack/lib/util.sh (hunk 1 above)
Comment (JoshuaAndrew) [reply hack/lib/util.sh:737]: Obviously bash 5.x can work normally, but this further narrows the optional range of compilation environment (eg. CentOS7 Ubuntu16.04), maybe bash 4.2 is the recommended value, better compatibility should also be one of the goals of the community.

Comment (dims) [reply hack/lib/util.sh]: @JoshuaAndrew this PR reflects reality, please feel free to follow up with changes to make things work in bash 4.2 

thanks!

**Comment by dims** [comment]
@justaugustus let's get this in please. (needs lgtm)

**Comment by justaugustus** [comment]
/lgtm
/approve

FYI @kubernetes/release-engineering

**Comment by borgerli** [comment]
We are still runningUbuntu 18.04(EOL in Apr 2023) with bash 4.4, which supports the "-v" test.  This PR make our CI/CD broken. 

Do we really need bash 5.x, since bash 4.2+ is OK with -v test?

**Comment by mrunalp** [comment]
@dims This also broke CRI-O CI running on RHEL 7.

**Pull request merged**